### PR TITLE
Add support for pdf in sub frames (https://github.com/electron/electr…

### PR DESCRIPTION
**Pull request by ahmedmohamedali**
This PR enable the PDF rendering in sub frame (Issue https://github.com/electron/electron/issues/9192). 
 It work with the chromium changes described in the PR https://github.com/electron/libchromiumcontent/pull/374.

### CODE REVIEW

File: atom/browser/atom_resource_dispatcher_host_delegate.cc [hunk 1]
```diff
@@ -134,22 +134,31 @@ AtomResourceDispatcherHostDelegate::CreateClientCertStore(
...
+	const std::string& mime_type,
+	GURL* origin,
+	std::string* payload) {
+	const content::ResourceRequestInfo* info =
+		content::ResourceRequestInfo::ForRequest(request);
+
+	int render_process_host_id;
+	int render_frame_id;
+	info->GetAssociatedRenderFrame(&render_process_host_id, &render_frame_id);
+	content::RenderFrameHost* rfh =
```
Comment (deepak1556) [inline atom/browser/atom_resource_dispatcher_host_delegate.cc]: `RenderFrameHost` methods should only be accessed on the UI thread.

File: atom/browser/atom_resource_dispatcher_host_delegate.cc [hunk 2]
```diff
@@ -134,22 +134,31 @@ AtomResourceDispatcherHostDelegate::CreateClientCertStore(
...
+	const base::FilePath& plugin_path,
+	const std::string& mime_type,
+	GURL* origin,
+	std::string* payload) {
+	const content::ResourceRequestInfo* info =
+		content::ResourceRequestInfo::ForRequest(request);
+
+	int render_process_host_id;
+	int render_frame_id;
+	info->GetAssociatedRenderFrame(&render_process_host_id, &render_frame_id);
```
Comment (deepak1556) [inline atom/browser/atom_resource_dispatcher_host_delegate.cc]: Better to verify the result of `GetAssociatedRenderFrame` method before proceeding with retrieving the `frame_tree_node_id`.

**Comment by ahmedmohamedali** [comment]
Hi @deepak1556,
Can you please review the changes I made after your first review ? 
Thanks.

**Comment by deepak1556** [comment]
Can you also add tests to https://github.com/electron/electron/blob/master/spec/chromium-spec.js#L979 ? Thanks!

**Comment by MarshallOfSound** [comment]
Looks like line endings were brutalized in this file 🤔

**Comment by ahmedmohamedali** [comment]
@deepak1556 ,

I'm not familiar with electron test environment. Can you tell me how to run the tests or point me to a document explaining how to run them ? Thanks

@MarshallOfSound ,

I have formatted the file with clang-format as suggested by @deepak1556. 
It isn't enough ?  Thanks.

**Comment by deepak1556** [comment]
@ahmedmohamedali you can look at https://github.com/electron/electron/blob/master/docs/development/build-system-overview.md#tests for how to run the tests.

**Comment by ckerr** [comment]
Changing the linefeeds makes the entire file show up in the diff.

**Comment by ahmedmohamedali** [comment]
thanks @ckerr,

I tried to fix the linefeeds with notepad++ but I  found it is the git commit  which is converting LF back to CRLF in windows.

**Comment by ahmedmohamedali** [comment]
Thanks @deepak1556.  I have added the following two test cases
- PDF in a sub frame
- PDF in a nested sub frame

**Comment by ahmedmohamedali** [comment]
Hi @deepak1556,
Is there anything missing for going forward with this PR ? Thanks.

**Comment by ahmedmohamedali** [comment]
Closing and reopening this PR to trigger CI builds

File: spec/fixtures/pages/pdf-in-nested-iframe.html [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<html>
+<body>
+<iframe id='outer-frame' src="./pdf-in-iframe.html"/>
+</body>
+</html>
+
+
```
Comment (alexeykuzmin) [inline spec/fixtures/pages/pdf-in-nested-iframe.html]: Empty lines are not needed here. In this file and in the html file above.
We should probably have a linter doing this kind of checks =/

File: spec/fixtures/pages/pdf-in-nested-iframe.html [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<html>
+<body>
+<iframe id='outer-frame' src="./pdf-in-iframe.html"/>
```
Comment (alexeykuzmin) [inline spec/fixtures/pages/pdf-in-nested-iframe.html]: Please use double quotes for attributes.

File: spec/chromium-spec.js [hunk 5]
```diff
@@ -1012,8 +1034,8 @@ describe('chromium feature', () => {
       w.webContents.loadURL(pdfSource)
     })
 
-    it('should not open when pdf is requested as sub resource', (done) => {
-      createBrowserWindow({plugins: true})
+    it('should not open when pdf is requested as sub resource', function (done) {
```
Comment (alexeykuzmin) [inline spec/chromium-spec.js]: Why `(done) =>` was changed to `function (done)`?

File: spec/chromium-spec.js [hunk 6]
```diff
@@ -955,19 +955,41 @@ describe('chromium feature', () => {
...
-      ipcMain.once('pdf-loaded', (event, state) => {
+    function testPDFIsLoadedInSubFrame (page, preloadFile, done) {
+      const pagePath = url.format({
+        pathname: path.join(fixtures, 'pages', page).replace(/\\/g, '/'),
+        protocol: 'file',
+        slashes: true
+      })
+
+      createBrowserWindow({plugins: true, preload: preloadFile})
+      ipcMain.once('pdf-loaded', function (event, state) {
```
Comment (alexeykuzmin) [inline spec/chromium-spec.js]: Is it possible that `'pdf-loaded'` listener will be called before the one of `'page-title-updated'`?

**Comment by alexeykuzmin** [comment]
> Now that electron/libchromiumcontent#374 is merged, the tests should pass.

@ahmedmohamedali, you need to update `vendor/libchromiumcontent` submodule reference to use a new libchromiumcontent revision.

File: spec/chromium-spec.js (hunk 6 above)
Comment (deepak1556) [reply spec/chromium-spec.js]: Its possible if the title is updated in javascript. But it shouldn't happen for this spec.

File: spec/chromium-spec.js (hunk 5 above)
Comment (ahmedmohamedali) [reply spec/chromium-spec.js]: @alexeykuzmin,
```function (done)``` was removed after my fork with the following change
[WIP] Upgrade more specs (#10945)
https://github.com/electron/electron/commit/e4214a6cbe084f6e038d955ae0a61df9394cd486

I think I have also missed it during the conflict resolution. Will fix it.

**Comment by zcbenz** [comment]
We are having some problems with libchromiumcontent binaries not being uploaded, so the tests are currently failing because of that.

**Pull request merged**